docs/src/pages/components/data-grid/editing/ConditionalValidationGrid.js

```diff
@@ -73,33 +47,61 @@ const rows = [
     price: randomPrice(0, 1000),
     dueAt: new Date(2021, 7, 4),
     isPaid: true,
-    paidAt: new Date(2021, 7, 2),
+    paymentMethod: 'Wire transfer',
   },
 ];
 
 export default function ConditionalValidationGrid() {
   const classes = useStyles();
-  const [editRowsModel, setEditRowsModel] = React.useState({});
+  const apiRef = useGridApiRef();
 
-  const handleEditRowsModelChange = React.useCallback((newModel) => {
-    const updatedModel = { ...newModel };
-    Object.keys(updatedModel).forEach((id) => {
-      const hasError =
-        updatedModel[id].isPaid.value && !updatedModel[id].paidAt.value;
-      updatedModel[id].paidAt = { ...updatedModel[id].paidAt, error: hasError };
-    });
-    setEditRowsModel(updatedModel);
-  }, []);
+  const columns = [
```

Comment:
We should memoize it to give a good example to users.
Non-memoized rows and columns tends to create unwanted behaviors.

x other cases

Reply:
Not always. `React.useMemo` will only be useful if the component re-renders, e.g. as result of a state change. Besides that we're introducing the solution without first having the problem. We only recommend memoizing the columns, not the rows: https://mui.com/components/data-grid/columns/#column-definitions. In this specific demo, memoization is not necessary because it doesn't have state.

Reply:
> We only recommend memoizing the columns, not the rows: https://mui.com/components/data-grid/columns/#column-definitions.

We should recommend it for the rows, it re-applies the filter / sorting / selection so in real applications it is an heavy task/

>  In this specific demo, memoization is not necessary because it doesn't have state.

I think it is simpler to do it on all our demos.
Until now, we always generated them outside of the component, even if the component did not re-render. It was not necessary for the component to work correctly. Of course if the Grid never re-renders, it is not actually doing anything.
But it makes it very clear for our user that we rely on the rows and columns reference for updates. Whereas personally if I saw a demo passing an array generated in the render method without memoization, I would probably take for granted that it can change between renders without any problem.